### ISSUE TITLE
Git not working: missing /dev/null

### ISSUE THREAD

**Issue by Pendrokar** [git-for-windows/git#1016]
- [x] I was not able to find an [open](https://github.com/git-for-windows/git/issues?q=is%3Aopen) or [closed](https://github.com/git-for-windows/git/issues?q=is%3Aclosed) issue matching what I'm seeing

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options
fatal: open /dev/null or dup failed: No such file or directory

Checked manually:
2.11 64-bit
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
Windows 10 Pro 64-bit

$ cmd.exe /c ver
Microsoft Windows [Version 10.0.14986]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
$ cat /etc/install-options.txt
Path Option: Cmd
SSH Option: OpenSSH
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Disabled
Enable Symlinks: Disabled
Enable Builtin Difftool: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

Microsoft Windows Insider (Level - Slow)
Was working before. So this issue might be due to an Windows update by Microsoft.

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

MinTTy

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
Anything that refers to git.exe fails with the same message.
```
 - What did you expect to occur after running these commands?

git.exe running any command

 - What actually happened instead?

fatal: open /dev/null or dup failed: No such file or directory

 - If the problem was occurring with a specific repository, can you provide the
   URL to that repository to help us with testing?

First updated 2.10 to 2.11 using the official website. No change. Uninstalled Git for Windows, rebooted, installed 2.11 again. No change.

Environment variables seem to be set properly. Writing git in CMD does got to git.exe.

git-bash.exe closes itself.
git-cmd.exe works.
bash.exe throws two "bash: /dev/null: No such device or address" lines before allowing input.

Related issue may be #768, which was about bash not git itself. There the user found solution by themselves. I have no idea what "/dev/null" is and it sounds like a like a Unix/Linux term.

**Comment by dscho** [comment]
Can you verify that `echo hello > nul` works in a plain `cmd`? I saw some reports in a quick web search where that was not working because the "Null Device Driver" was either uninstalled or not working...

**Comment by Pendrokar** [comment]
```
>echo hello > nul
The system cannot find the file specified.
```
Still do not understand why the installer doesn't take care of this. The following method through using .inf file fails, says the method is out of date:
https://msdn.microsoft.com/en-us/windows/hardware/drivers/install/installing-a-null-driver

I had given up and used Windows 10's 'Bash on Ubuntu on Windows' environment for git.

**Comment by dscho** [comment]
> Still do not understand why the installer doesn't take care of this.

This is not a problem of Git for Windows, but one of your setup. `NUL` is not expected to be disabled, like, ever, hence Git for Windows' installer does not try to do anything about it.

I tried to find out what may have happened on your system (you will most likely run into more serious problems if you do not fix the issue that `NUL` cannot be found), but all I found was random (and rare) problems on Windows 7, not Windows 10.

In my Windows 10 setup, there is a `%WINDOWS%\system32\drivers\null.sys` (which is the correct driver) and the registry key `NUL` in `HKLM\SYSTEM\CurrentControlSet\Control\Session Manager\DOS Devices` is set to `\Device\Null`. Maybe something got messed up there in your setup.

> The following method through using .inf file fails, says the method is out of date:
> https://msdn.microsoft.com/en-us/windows/hardware/drivers/install/installing-a-null-driver

That explains how to install a NULL driver, not the NUL device driver (which is confusingly called the "Null Device Driver").

> I had given up and used Windows 10's 'Bash on Ubuntu on Windows' environment for git.

Good! I will close this ticket, then.
